overlord/snapstate/storehelpers.go

```diff
@@ -781,6 +800,46 @@ func refreshCandidates(ctx context.Context, st *state.State, names []string, rev
 	return updates, stateByInstanceName, ignoreValidationByInstanceName, nil
 }
 
+// SnapHolds returns a map of held snaps to lists of holding snaps (including
+// "system" for user holds).
+func SnapHolds(st *state.State, snaps []string) (map[string][]string, error) {
```

Comment:
right now both HeldSnaps and SnapHolds are exported, neither is so far used outside snapstate atm though. Does it make sense to keep exporting both?

Reply:
I didn't do the initial implementation maybe @MiguelPires has a more convincing reason, but I think the export was done to [facilitate testing](https://github.com/ZeyadYasser/snapd/commit/6e09eb12b61288eba5ab06faeed9fa89bd3cfe27#diff-8e164732575e20f0c572a4ff42ad8c71e885a6ecf24356bb31a27e5e6126d4e4R510) as testing package is `snapstate_test` instead of `snapstate`.

Reply:
we can solve this in a follow-up when @MiguelPires is back to discuss, it's not a blocker as such